clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -95,27 +95,54 @@ OffloadTargetInfo::OffloadTargetInfo(const StringRef Target,
                                      const OffloadBundlerConfig &BC)
     : BundlerConfig(BC) {
 
-  // <kind>-<triple>[-<target id>[:target features]]
-  // <triple> := <arch>-<vendor>-<os>-<env>
   SmallVector<StringRef, 6> Components;
   Target.split(Components, '-', /*MaxSplit=*/5);
-  assert((Components.size() == 5 || Components.size() == 6) &&
-         "malformed target string");
-
-  StringRef TargetIdWithFeature =
-      Components.size() == 6 ? Components.back() : "";
-  StringRef TargetId = TargetIdWithFeature.split(':').first;
-  if (!TargetId.empty() &&
-      clang::StringToOffloadArch(TargetId) != clang::OffloadArch::UNKNOWN)
-    this->TargetID = TargetIdWithFeature;
-  else
-    this->TargetID = "";
+  if (Components.size() == 5 || Components.size() == 6) {
```

Comment:
One minor nit: May be we can reorg this code as follows (to avoid merge issues)
if (Components.size() < 5) {
...// Do our thing
  return;
}
// Add community code as is.

Reply:
:) yes, that makes sense.  I had an earlier revision that was similar to that - and it is easier for merging.